### PR TITLE
Prompt user for username and password if required

### PR DESCRIPTION
**Pull request by kev24uk**
Hi,

I made a small change to prompt the user for a username and password since I needed it to use this on a private repository at work.

Feel free to merge it in if you think it's useful!

Thanks,
Kevin.

### CODE REVIEW

File: modules/svneverever/__main__.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 import os
 import sys
 import time
+import getpass
```
Comment (hartwork) [inline modules/svneverever/__main__.py]: There is no CI to respect `.pre-commit-config.yaml` yet, but this project uses isort and `getpass` would need to go first, according to isort.

If you want to see/use it yourself, please do this on the Git clone:
```shell
virtualenv --python=python3 .py3
source .py3/bin/activate
pip install pre-commit
pre-commit install --install-hooks
pre-commit run --all-files
```
[pre-commit](https://pre-commit.com/) will also check/sanitize future commits for you, then. Please give it a try, pre-commit rocks :smile:

File: modules/svneverever/__main__.py [hunk 2]
```diff
@@ -255,12 +256,17 @@ def command_line():
 
     return args
 
+def login(realm, username, may_save):
```
Comment (hartwork) [inline modules/svneverever/__main__.py]: Let's not expose this symbol, please use `_login` or so.

File: modules/svneverever/__main__.py [hunk 3]
```diff
@@ -255,12 +256,17 @@ def command_line():
...
+    username = six.input("Please enter your username: ")
+    password = getpass.getpass()
```
Comment (hartwork) [inline modules/svneverever/__main__.py]: The default prompt of `getpass` is `Password: `. Let's make these two match more regarding output style, either
```
username = six.input('Username: ')
password = getpass.getpass()
```
or even
```
username = six.input('Username: ')
password = getpass.getpass('Password: ')
```

File: modules/svneverever/__main__.py [hunk 4]
```diff
@@ -255,12 +256,17 @@ def command_line():
...
+    return True, username, password, False
```
Comment (hartwork) [inline modules/svneverever/__main__.py]: :+1: Thumbs up on that `False` for don't-save at the end, every nice!

File: modules/svneverever/__main__.py (hunk 3 above)
Comment (hartwork) [inline modules/svneverever/__main__.py]: I think we need some error handling in `login` as well, e.g. `KeyboardInterrupt` or `EOFError` during `getpass`

File: modules/svneverever/__main__.py [hunk 5]
```diff
@@ -255,12 +256,17 @@ def command_line():
...
 
 def main():
     args = command_line()
 
     # Build tree from repo
     client = pysvn.Client()
+    client.callback_get_login = login
```
Comment (hartwork) [inline modules/svneverever/__main__.py]: It would be nice to guard this line by a new command line switch, either `--interactive` or `--non-interactive`.

**Comment by kev24uk** [comment]
I've pushed the requested changes, good ideas! Hopefully I've got the gist of what you were after. I'd not used pre-commit before, I'm used to working with C# which provides code analysis so this is similar, like it!

**Comment by hartwork** [comment]
I'm still unsure about a few details. I'll try to setup a password protected SVN repository following https://stackoverflow.com/a/11810992/11626624 to experiment myself in the next few days.

**Comment by hartwork** [comment]
Hi @kev24uk ,

I found some time to play with this myself now. I added a few small related improvements on top myself and released the result as 1.4.0 here and on PyPI just now. I hope you are happy with the result. Please report bugs if you run into and. Thanks for your contribution!

Best, Sebastian

**Pull request merged**